pkg/definition/definition.go

```diff
@@ -375,7 +377,7 @@ func ValidDefinitionTypes() []string {
 }
 
 // SearchDefinition search the Definition in k8s by traversing all possible results across types or namespaces
-func SearchDefinition(definitionName string, c client.Client, definitionType string, namespace string) ([]unstructured.Unstructured, error) {
+func SearchDefinition(definitionName string, c client.Client, definitionType string, namespace string, addonFilter string) ([]unstructured.Unstructured, error) {
```

Comment:
the addonFilter can be a function interface, then it's portable to add other filter for extensibility:

```suggestion
func SearchDefinition(definitionName string, c client.Client, definitionType, namespace string, addonFilter func(unstructured.Unstructured)bool) ([]unstructured.Unstructured, error) {
```

Reply:
Great, I refactored the code to accept function interface as generic filters, and encapsulated the original implementation into one of the filters.

Reply:
Well, in the original function parameters `SearchDefinition(definitionName string, c client.Client, definitionType, namespace string)`, like `definitionType`, `namespace`, and `definitionName` can all be replaced with filters, in unified manner. I wonder if this refactor is necessary.

Reply:
@charlie0129 You're really smart! That's exactly what I mean. Please make sure they all covered by tests.